core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -1143,17 +1154,123 @@ class GroupCoordinatorTest {
       initialRebalanceResult.generation + 1,
       Set(leaderInstanceId, followerInstanceId, newMemberInstanceId),
       groupId,
-      CompletingRebalance)
+      CompletingRebalance,
+      Some(protocolType))
 
     checkJoinGroupResult(newFollowerResult,
       Errors.NONE,
       initialRebalanceResult.generation + 1,
       Set.empty,
       groupId,
       CompletingRebalance,
+      Some(protocolType),
       expectedLeaderId = newLeaderResult.memberId)
   }
 
+  @Test
+  def testJoinGroupReturnsAnNoneProtocolTypeWhenAnErrorOccurs(): Unit = {
```

Comment:
nit: "an none" seems ungrammatical

Reply:
Indeed. I am not sure how I managed to write this ;)